gapil/runtime/cc/map.inc

```diff
@@ -317,7 +317,7 @@ void Map<K, V, DENSE>::Allocation::remove(K key) {
 template<typename K, typename V, bool DENSE>
 void Map<K, V, DENSE>::Allocation::clear() {
     auto elems = els();
-    for (uint64_t i = 0; i < capacity; ++i) {
+    for (uint64_t i = 0; i < capacity && count > 0; ++i) {
         switch(elems[i].used) {
```

Comment:
Bug not introduced in these changes, but shouldn't we be setting the `used` flag to `GAPIL_MAP_ELEMENT_EMPTY` for each element? Seems like we could easily double destruct with something like:

```
map[3] = obj_a;
map.clear();
map[5] = obj_b;
map.clear(); // obj_a destructed again
```
(note `5 < GAPIL_MIN_MAP_SIZE`)

Your `memset` in `clear_keep` looks like it would fix this.

Reply:
clear() actually deletes the elements array.
However one of my previous optimizations breaks this though. (GAPIL_MIN_MAP_SIZE) I think we need an else() memset there.

Reply:
Yeah, I see that the elements are deleted if the count is greater than GAPIL_MIN_MAP_SIZE. The example is below this threshold and triggers the bug described.